File: ruby-pr/53/ruby/code-quality-readability-maintainability/readable_data_transformer.rb
class ReadableDataTransformer
  def initialize
    # {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
    @transformations = {
      'upper' => :upcase,
      'lower' => :downcase,
      'trim' => :strip,
      'reverse' => :reverse,
      'capitalize' => method(:capitalize_string)
    }
    # {/fact}
  end

  def transform(input, operation)
    transformation = @transformations[operation]
    return input unless transformation # Return original if operation not supported

    input.map { |item| apply_transformation(item, transformation) }
  end

  private

  def apply_transformation(item, transformation)
    if transformation.is_a?(Symbol)
      item.send(transformation)
    else
      transformation.call(item)
    end
  end

  def capitalize_string(input)
    input.downcase.capitalize
  end
end

File: php-pr/51/php/code-quality-readability-maintainability/ReadableDataTransformer.php
<?php

class ReadableDataTransformer
{
    private $transformations;

    public function __construct()
    {
        // {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
        $this->transformations = [
            'upper' => 'strtoupper',
            'lower' => 'strtolower',
            'trim' => 'trim',
            'reverse' => 'strrev',
            'capitalize' => [$this, 'capitalizeString']
        ];
        // {/fact}
    }

    public function transform($input, $operation)
    {
        if (!isset($this->transformations[$operation])) {
            return $input; // Return original if operation not supported
        }

        $transformation = $this->transformations[$operation];
        return array_map($transformation, $input);
    }

    private function capitalizeString($input)
    {
        return ucfirst(strtolower($input));
    }
}